### PR TITLE
change test-cmd to be silent

### PR DESCRIPTION
**Pull request by stevekuznetsov**
cc @liggitt 
This follows the template in `hack/test-integration` for buckets in `hack/test-cmd` to reduce output spam.

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@liggitt Currently seeing this, need help understanding why the output is not caught: https://gist.github.com/stevekuznetsov/3e2147c0d18a379a48cd

File: hack/test-cmd.sh [hunk 1]
```diff
@@ -199,85 +242,98 @@ if [[ "${API_SCHEME}" == "https" ]]; then
...
-oadm policy add-cluster-role-to-user system:image-pruner pruner --config="${MASTER_CONFIG_DIR}/admin.kubeconfig"
-oc login --server=${KUBERNETES_MASTER}/ --certificate-authority="${MASTER_CONFIG_DIR}/ca.crt" -u pruner -p anything
-# this shouldn't fail but instead output "Dry run enabled - no modifications will be made. Add --confirm to remove images"
-oadm prune images
+exectest "login" test_login_logout
 
 # log in and set project to use from now on
 oc login --server=${KUBERNETES_MASTER} --certificate-authority="${MASTER_CONFIG_DIR}/ca.crt" -u test-user -p anything
-oc get projects
-oc project project-foo
```
Comment (stevekuznetsov) [inline hack/test-cmd.sh]: @liggitt  Just making sure that this line and the one above was not ever used.

**Comment by stevekuznetsov** [comment]
Current output: https://gist.github.com/stevekuznetsov/2fe3bdabc94a2310b81b
I am not sure the normal user wants to see 350 lines of `pprof` output on their local machine. We can make this a flag that is set on the CI servers if it's useful there.

**Comment by stevekuznetsov** [comment]
`$ hack/test-cmd.sh`: [output](https://gist.github.com/stevekuznetsov/8cdb57e0c4c78c2b5dbf):
```

$ hack/test-cmd.sh 
openshift: openshift v1.0.6-318-gf934c42-dirty kubernetes v1.1.0-alpha.1-653-g86b4e77
[INFO] Create certificates for the OpenShift server to /tmp/openshift-cmd/PnHp/openshift.local.config/master
Generated new key pair as /tmp/openshift-cmd/PnHp/openshift.local.config/master/serviceaccounts.public.key and /tmp/openshift-cmd/PnHp/openshift.local.config/master/serviceaccounts.private.key
Generating node credentials ...
Created node config for 127.0.0.1 in /tmp/openshift-cmd/PnHp/openshift.local.config/node-127.0.0.1
Wrote master config to: /tmp/openshift-cmd/PnHp/openshift.local.config/master/master-config.yaml
Generating node credentials ...
Created node config for 127.0.0.1 in /tmp/openshift-cmd/PnHp/openshift.local.config/node-127.0.0.1

Running tests...
ok      validate
ok      apiserver
ok      atomic
ok      login
ok      whoami
ok      config
Flag shorthand -t has been deprecated, please use --template instead
ok      edit
ok      export
ok      basicresources
ok      images
ok      secrets
ok      help
ok      templates
ok      policy
ok      deployments
ok      builds
ok      admin
ok      volumes
ok      newapp
ok      metrics

Complete
hack/test-cmd.sh took 333 seconds
```
`$ PROFILE=true hack/test-cmd.sh`: [output](https://gist.github.com/stevekuznetsov/b779fee6b26a50e97a30)
`$ VERBOSE=true PROFILE=true hack/test-cmd.sh`: [output](https://gist.github.com/stevekuznetsov/7fccb2c4ee658c024f8b)

When `$VERBOSE` and `$PROFILE` are both set, `test-cmd` irreducibly fails  ... investigating this now.

File: hack/test-cmd.sh [hunk 2]
```diff
@@ -45,6 +50,36 @@ function cleanup()
 trap "exit" INT TERM
 trap "cleanup" EXIT
 
+function exectest() {
+  testname=$1
+  testcmd=$2
+  echo "Running $testname..."
+
+  result=1
+  if [ -n "${VERBOSE-}" ]; then
```
Comment (Miciah) [inline hack/test-cmd.sh]: I think this is backwards: `-n "${VERBOSE-}"` means `$VERBOSE` is not empty, which means it should be verbose, right?

Comment (liggitt) [inline hack/test-cmd.sh]: right, in which case, we invoke it directly rather than capturing the output

**Comment by stevekuznetsov** [comment]
@liggitt @smarterclayton thoughts?

**Comment by smarterclayton** [comment]
I might suggest redirecting stdout to a file.  And if the thing fails, print the last X lines of the file.  And then upload the file.  Instead of changing every line of hack/test-cmd.sh

**Comment by stevekuznetsov** [comment]
@smarterclayton I don't really think that's user-friendly. We can do the redirection when the call to `test-cmd` is made. Printing output from the tests that failed instead of some arbitrary `tail` is also somewhat more robust. When later tests fail, the last thousand or ten thousand lines are the server output, not the test output, so we'd have to choose some ridiculous number or somehow have it scale with the test output, which seems convoluted.

**Comment by smarterclayton** [comment]
I've never seen test-cmd fail where the error wasn't in the last 20 lines.
If the artifacts on the server contain the whole log, even better.

On Oct 6, 2015, at 4:58 PM, Steve Kuznetsov <notifications@github.com>
wrote:

@smarterclayton <https://github.com/smarterclayton> I don't really think
that's user-friendly. We can do the redirection when the call to test-cmd
is made. Printing output from the tests that failed instead of some
arbitrary tail is also somewhat more robust. When later tests fail, the
last thousand or ten thousand lines are the server output, not the test
output, so we'd have to choose some ridiculous number or somehow have it
scale with the test output, which seems convoluted.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4893#issuecomment-145998886>.

**Comment by smarterclayton** [comment]
And server output should be displayed to end users 90% of the time - it
almost never has anything useful.

On Oct 6, 2015, at 4:58 PM, Steve Kuznetsov <notifications@github.com>
wrote:

@smarterclayton <https://github.com/smarterclayton> I don't really think
that's user-friendly. We can do the redirection when the call to test-cmd
is made. Printing output from the tests that failed instead of some
arbitrary tail is also somewhat more robust. When later tests fail, the
last thousand or ten thousand lines are the server output, not the test
output, so we'd have to choose some ridiculous number or somehow have it
scale with the test output, which seems convoluted.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4893#issuecomment-145998886>.

**Comment by deads2k** [comment]
> I might suggest redirecting stdout to a file. And if the thing fails, print the last X lines of the file. And then upload the file. Instead of changing every line of hack/test-cmd.sh

I like this idea a lot.  I have access to all the information if I really need, but very rarely do I care about what happened even 10 lines past the bottom.

**Comment by danmcp** [comment]
@stevekuznetsov Needs rebase

**Comment by stevekuznetsov** [comment]
@smarterclayton Where should the output files (server log, verbose output log, profiling log) live?

**Comment by smarterclayton** [comment]
See my most recent changes, but same location we put Openshift.log

On Oct 21, 2015, at 11:24 AM, Steve Kuznetsov <notifications@github.com>
wrote:

@smarterclayton <https://github.com/smarterclayton> Where should the output
files (server log, verbose output log, profiling log) live?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4893#issuecomment-149930828>.

**Comment by stevekuznetsov** [comment]
This is going to be subsumed by https://github.com/openshift/origin/pull/5719 and following pulls. Closing.

**Pull request closed without merging**